android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java

```diff
@@ -1,93 +1,67 @@
-        if (eventDateStringLivePair == null) {
-            eventDateStringLivePair = new MutableLiveData<>();
-        }
-        if (eventDateStringData == null) {
-            eventDateStringData = new ArrayList<>();
-            eventDatesRealmResults = realmRepo.getEventDates();
-            addEventDatesRealmResultsChangeListener();
-        } else {
-            for (int i = 0; i < eventDateStringData.size(); i++) {
-                eventDateStringLivePair.setValue(eventDateStringData.get(i));
-            }
-        }
+    public LiveData<List<Pair<String, String>>> getEventDateString() {
+        if (eventDateStringLivePair == null)
+            subscribeToEventDateString();
         return eventDateStringLivePair;
     }
 
-    private void addEventDatesRealmResultsChangeListener() {
-        eventDatesRealmResults.addChangeListener((eventDates, orderedCollectionChangeSet) -> {
-            eventDateStringData.clear();
+    private void subscribeToEventDateString() {
+        LiveRealmData<EventDates> eventDatesLiveRealmData = RealmDataRepository.asLiveData(realmRepo.getEventDates());
+        eventDateStringLivePair = Transformations.map(eventDatesLiveRealmData, eventDates -> {
             int eventDays = eventDates.size();
+            List<Pair<String, String>> eventDateStringData = new ArrayList<>();
             for (int i = 0; i < eventDays; i++) {
                 String date = eventDates.get(i).getDate();
                 try {
                     String formattedDate = DateConverter.formatDay(date);
-                    Pair<String,String> data = new Pair<>(formattedDate, date);
+                    Pair<String, String> data = new Pair<>(formattedDate, date);
                     eventDateStringData.add(data);
-                    eventDateStringLivePair.setValue(data);
                 } catch (DateTimeParseException pe) {
                     Timber.e(pe);
                     Timber.e("Invalid date %s in database", date);
                 }
             }
+            return eventDateStringData;
         });
     }
 
     public LiveData<List<Track>> getTracks() {
         if (tracksLiveData == null) {
-            tracksLiveData = new MutableLiveData<>();
-            tracksRealmResults = realmRepo.getTracks();
-            tracksRealmResults.addChangeListener((tracks, orderedCollectionChangeSet) -> {
-                tracksLiveData.setValue(tracks);
-            });
+            subscribeToTracks();
         }
         return tracksLiveData;
     }
 
-    private void clearListeners() {
-        if (eventDatesRealmResults != null) {
-            eventDatesRealmResults.removeAllChangeListeners();
-        }
-        if (tracksRealmResults != null) {
-            tracksRealmResults.removeAllChangeListeners();
-        }
-    }
-
-    @Override
-    protected void onCleared() {
-        clearListeners();
-        super.onCleared();
+    private void subscribeToTracks() {
```

Comment:
Use RealmDataWrapper

Reply:
@iamareebjamal Sorry didn't get you?

Reply:
There are new helper classes in the project which abstract the functionality of managing Realm connections, please use those